src/controllers/auth.controller.ts

```diff
@@ -5,25 +5,31 @@ import {
   BodyParam,
   UnauthorizedError,
   BadRequestError,
-  Res
+  Req,
+  Res,
+  Authorized
 } from 'routing-controllers';
 import omit from 'lodash/omit';
 import { Service } from 'typedi';
 import { User } from '@entities/user.entity';
 import { SessionService } from '@services/session.service';
 import { Errors } from '@constants/errorMessages';
+import { Request, Response } from 'express';
 
 @JsonController('/auth')
 @Service()
 export class AuthController {
   constructor(private readonly sessionService: SessionService) {}
 
   @Post('/signup')
-  async signUp(@Body({ validate: false }) user: User, @Res() response: any) {
+  async signUp(
+    @Body({ validate: false }) user: User,
+    @Res() response: Response
+  ) {
     try {
       const newUser = await this.sessionService.signUp(user);
       return response.send(omit(newUser, ['password']));
-    } catch (error) {
+    } catch (error: any) {
```

Comment:
Could we have a generic interface or type that identifies an Error? so that we can get rid of the `any`

Reply:
![image](https://user-images.githubusercontent.com/58518912/123301218-cdce7a80-d4f1-11eb-945e-054a7bb38472.png)
I was researching and ts does not allow this. We could check inside the catch block the type of error and then execute different actions, but not in the way of defining the type of the catch argument.

Reply:
@zk182 FYI about https://github.com/rootstrap/node-ts-api-base/pull/550#discussion_r665012679